private-media-framework/draft-ietf-perc-private-media-framework.md

```diff
@@ -548,37 +558,53 @@ re-keyed is outside the scope of this document, but this framework
 does accommodate re-keying during the life of a conference.
 
 When a Key Distributor decides to re-key a conference, it transmits a
-specific message defined in PERC EKT [@!I-D.ietf-perc-srtp-ekt-diet] to
-each of the conference participants.  The endpoint **MUST** create a
+new EKTKey message [@!I-D.ietf-perc-srtp-ekt-diet] to
+each of the conference participants containing the new EKT Key.
+Upon receipt of the new EKT Key, the endpoint **MUST** create a
 new SRTP master key and prepare to send that key inside a Full EKT
-Field using the new EKTKey.  Since it may take some time for all of the
-endpoints in conference to finish re-keying, senders **MUST** delay a
-short period of time before sending media encrypted with the new
-master key, but it **MUST** be prepared to make use of the information
-from a new inbound EKT Key immediately. See Section 2.2.2 of
-[@!I-D.ietf-perc-srtp-ekt-diet].
+Field using the new EKT Key as per Section 4.5 of [@!I-D.ietf-perc-srtp-ekt-diet].
+Since it may take some time for all of the endpoints in conference to finish
+re-keying, senders **MUST** delay a short period of time before sending media
+encrypted with the new master key and before utilizing the new EKT Key
+(continuing using the old EKT Key temporarily), but it **MUST** be prepared to
+process packets containing the new EKT Key immediately. The EKT SPI field can
+be used to differentiate between tags encrypted with the old and new keys.
+
+> Editor's Note: Now much time is a "short period of time"? It would be
+  really nice if the EKTKey message contained this delay. And the KD knows
+  roughly how long it would take to send out messages. For a two-party
+  conference, this might be 100ms, but for a large conference it might be
+  several seconds.
+
+After re-keying, an endpoint should retain old SRTP master keys and EKT Key
+for a period of time for the purpose of ensuring it can decrypt packets.
+The time period to retain old keys is not specified, but they should be
+retained at least for the period of time required to re-key the conference
+or handle late-arriving or out-of-order packets.  A period of 60s should be
+considered a generous retention period, but an endpoint **MAY** retain old
+keys until the end of the conference.
```

Comment:
I don't think Ben had an issue with this point on the receiving side, so the elaboration and the 60s mention is unwarranted, IMHO.   How about ...

"After re-keying, an endpoint **SHOULD** retain prior SRTP master keys and EKT Key for a period of time sufficient for the purpose of ensuring it can decrypt late-arriving or out-of-order packets or packets sent by other endpoints that used the prior keys for a period of time after re-keying began.   An endpoint **MAY** retain old
keys until the end of the conference."